manim/scene/scene.py

```diff
@@ -28,28 +28,34 @@
 class Scene(Container):
     """A Scene is the canvas of your animation.
 
-    All of your own named Scenes will be subclasses of Scene, or other named
-    scenes.
+    The primary role of :class:`Scene` is to provide the user with tools to manage
+    mobjects and animations.  Generally speaking, a manim script consists of a class
+    that derives from :class:`Scene` whose :meth:`Scene.construct` method is overriden
+    by the user's code.
+
+    Mobjects are displayed on screen by calling :meth:`Scene.add` and removed from
+    screen by calling :meth:`Scene.remove`.  All mobjects currently on screen are kept
+    in :attr:`Scene.mobjects`.  Animations are played by calling :meth:`Scene.play`.
+
+    A :class:`Scene` is rendered internally by calling :meth:`Scene.render`.  This in
+    turn calls :meth:`Scene.setup`, :meth:`Scene.construct`, and
+    :meth:`Scene.tear_down`, in that order.
+
+    It is not recommended to override the ``__init__`` method in user Scenes.  For code
+    that should be ran before a Scene is rendered, use :meth:`Scene.setup` instead.
+
 
     Examples
     --------
-    Override the construct() method to tell Manim what should go on in the
-    Scene.
+    Override the :meth:`Scene.construct` method with your code.
 
     .. code-block:: python
 
         class MyScene(Scene):
             def construct(self):
-                self.play(
-                    Write(Text("Hello World!"))
-                )
-
-    Some important variables to note are:
-        camera: The camera object to be used for the scene.
-        file_writer : The object that writes the animations in the scene to a video file.
-        mobjects : The list of mobjects present in the scene.
-        foreground_mobjects : List of mobjects explicitly in the foreground.
-        random_seed: The seed with which all random operations are done.
+                t = Text("Hello World!")
+                self.add(t)
+                self.play(Write(t))
```

Comment:
```suggestion
                t1 = Text("Hello ")
                t2 = Text("World!").next_to(t1,RIGHT)
                self.add(t1)
                self.play(Write(t2))
```

Reply:
Why? I would rather keep the example simple and to the point.  Since this is for `Scene`, the point is "this is how you create a scene" and not "here is a cool thing you can do with Text"

Reply:
As mentioned in the next comment, things like
```python
class ABC(Scene):
    def construct(self):
        t = Text("Hello World!")
        self.add(t)
        self.wait()
        self.play(Write(t))
```
could lead to unintended behaviour

Reply:
I'm not following - what comment? What unintended behavior? :sweat_smile:

Reply:
See this script:
```python
class ABC(Scene):
    def construct(self):
        t = Text("Hello World!")
        self.wait()
        self.add(t)
        self.wait()
        self.play(Write(t))
```
It once appears, then disappears, then writes.
![ABC](https://user-images.githubusercontent.com/44469195/100758735-22909380-33f0-11eb-84d0-9aba2a771525.gif)

Reply:
Oh! Right! Then I'd rather do just `self.play(Write(Text("Hello World!")))`. Is that okay?

Reply:
sounds good!

Reply:
Done!